### DISCUSSION TITLE
FutureProvider doesn't work as intended, execution pauses on uncaught exception

### DISCUSSION THREAD

**Discussion by Tobi696** [rrousselGit/riverpod#1574] [Q&A]
I have following FutureProvider:
```dart
final categoryProvider = FutureProvider.family.autoDispose<CategoriesIdGet200Response, int>((ref, id) async {
  final api = ref.read(apiProvider);
  final res = (await api.getCategoriesApi().categoriesIdGet(id: id)).data!;
  return res;
});
```

In my widget, I use it like this:
```dart
final category = ref.watch(categoryProvider(id));
category.when(
      data: (category) {
             return Column(children: [...]);
      },
      error: (e, _) {
        showErrorSnackBar(theme, e);
        return const SizedBox();
      },
      loading: () => const Center(child: CircularProgressIndicator()),
    );
```

However, when the future awaited in the provider throws an exception, vscode pauses debugging, shows me the exception and even when I continue debugging, the error Builder of the AsyncValue is actually never used.
This is the Exception that is thrown: `DioError (DioError [DioErrorType.response]: Http status error [500]`

Any help is appreciated!

**Comment by rrousselGit** [comment]
You can't call "showErrorSnackbar"  within the build method like you did

**Comment by Tobi696** [comment]
Ok thank you, so that fixes the setState called during build problem, but it still makes execution pause on an uncaught exception. Do you have any tip on how to fix that apart from disabling breakpoints on uncaught exceptions in vscode?

**Reply by rrousselGit** [reply]
It's not a bug. You do have an uncaught exception 😊

Riverpod explicitly reports them. Otherwise debugging would be difficult

**Reply by Tobi696** [reply]
Alright, thanks for your help!
